src/sst/core/baseComponent.h

```diff
@@ -329,6 +349,18 @@ class BaseComponent {
         return registerStatistic<T>(std::string(statName), std::string(statSubId));
     }
 
+    /** Called by the Components and Subcomponent to perform a statistic Output.
+      * @param stat - Pointer to the statistic.
+      * @param EndOfSimFlag - Indicates that the output is occurring at the end of simulation.
+      */
+    void performStatisticOutput(StatisticBase* stat, bool endOfSimFlag = false);
+
+    /** Performs a global statistic Output.
+     * This routine will force ALL Components and Subcomponents to output their statistic information.
+     * This may lead to unexpected results if the statistic counts or data is reset on output.
+     * @param endOfSimFlag - Indicates that the output is occurring at the end of simulation.
+     */
+    void performGlobalStatisticOutput(bool endOfSimFlag = false);
```

Comment:
Unless it's been fixed, this function is still not truly global - does not work across threads/ranks. I do not think it's pressing that it be fixed immediately, but it needs to get put in an issue and/or noted in the comment here.

Reply:
Okay, we should open an issue.  This just calls the same function in the stats engine.  I just didn't like having to give a handle to the engine to a component.  I don't think we want to try to fix it before the 10.0 release since getting a truly global stats dump will challenging.  This is only used in one place in the elements repository, do we want to just remove the capability from elements?

Reply:
No, Ariel and Ember need the capability. I think we just need to open an issue since this exposes a partially broken function more directly to baseComponents and fix it as soon as possible after 10.0.